src/sink/send_error.rs

```diff
@@ -0,0 +1,35 @@
+use std::fmt;
+use std::error::Error;
+use std::any::Any;
+
+/// Error type for sending, used when the receiving end of a channel is
+/// dropped
+pub struct SendError<T>(pub T);
```

Comment:
Despite this being the same across the two modules I think I'd prefer to keep separate `SendError` types for now if possible. That way we can customize them (if need be) and then perhaps later in 0.2 or such if it's required we can unify them.

In the meantime though I don't think we get *too* much benefit from unifying the types, other than a little deduplication in this crate.